examples/experimental/dagster-airlift/examples/dbt-example/dbt_example/dagster_defs/definitions.py

```diff
@@ -33,10 +26,20 @@ def dbt_project_path() -> Path:
 defs = build_defs_from_airflow_instance(
     airflow_instance=airflow_instance,
     orchestrated_defs=defs_from_factories(
-        PythonDefs(
+        CSVToDuckdbDefs(
```

Comment:
The main purpose of the custom `DefsFactory` here is to produce an API/user code matching the custom Airflow operator, is that right? e.g. there's no technical reason that there needs to be a 1:1 mapping of operator to defs factory - just making sure I understand this right.

It seems like a user could just use a `PythonDefs` here with an inline fn/lambda to define the inputs to `load_csv_to_duckdb`, but the downside is that there's a higher mental load in migrating that way since the user code ends up looking less similar moving airflow->Dagster?

Reply:
You're correct that PythonDefs could be used here, in the same way that a PythonOperator could be used to easily achieve the same goal of the load_csv_etc operator we use. The goal is to make it explicitly clear this semi 1:1 mapping exists for your custom operators / provide a clear path forward. The example is contrived, but the pattern isn't

Reply:
Thanks, that makes sense 👍